engine/execution/storehouse/in_memory_register_store.go

```diff
@@ -0,0 +1,336 @@
+package storehouse
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/engine/execution"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+var _ execution.InMemoryRegisterStore = (*InMemoryRegisterStore)(nil)
+
+var ErrNotExecuted = fmt.Errorf("block is not executed")
+
+type ErrPruned struct {
```

Comment:
nit: should this be named `PrunedError` since it implements a new error type, whereas ErrPruned would normally be just an existing error value?